### PR TITLE
Pass functions by "reference" to bis event handlers

### PR DESCRIPTION
**Pull request by PabstMirror**
Thanks to a discovery from @dedmen  (ref https://github.com/CBATeam/CBA_A3/pull/588)

Passing a large code function to a BIS EH has a significant performance cost.

Using interact_menu as an example.
```
addMissionEventHandler ["Draw3D", {x3 = diag_tickTime}];
for "_index" from 1 to 2000 do {
    // addMissionEventHandler ["Draw3D", FUNC(render)]; // (660 ms / 2000) = .33 ms
    addMissionEventHandler ["Draw3D", {call FUNC(render)}]; // (28 ms / 2000) = 0.014 ms
};
addMissionEventHandler ["Draw3D", {diag_log text format ["2000 calls took %1 ms", 1000 * (diag_tickTime - x3)]}];
```

So an extra processing time of ~0.3 ms when passing the entire function vs passing a short bit of code that calls the function.

### CODE REVIEW

File: addons/hearing/XEH_postInit.sqf [hunk 1]
```diff
@@ -60,9 +60,9 @@ GVAR(lastPlayerVehicle) = objNull;
         };
         // Don't add a new EH if the unit respawned
         if ((_player getVariable [QGVAR(firedEH), -1]) == -1) then {
-            private _firedEH = _player addEventHandler ["FiredNear", LINKFUNC(firedNear)];
+            private _firedEH = _player addEventHandler ["FiredNear", {call (firedNear)}];
             _player setVariable [QGVAR(firedEH), _firedEH];
-            private _explosionEH = _player addEventHandler ["Explosion", LINKFUNC(explosionNear)];
+            private _explosionEH = _player addEventHandler ["Explosion", {call (explosionNear)}];
```
Comment (commy2) [inline addons/hearing/XEH_postInit.sqf]: ?

File: addons/hearing/XEH_postInit.sqf [hunk 2]
```diff
@@ -60,9 +60,9 @@ GVAR(lastPlayerVehicle) = objNull;
         };
         // Don't add a new EH if the unit respawned
         if ((_player getVariable [QGVAR(firedEH), -1]) == -1) then {
-            private _firedEH = _player addEventHandler ["FiredNear", LINKFUNC(firedNear)];
+            private _firedEH = _player addEventHandler ["FiredNear", {call (firedNear)}];
```
Comment (commy2) [inline addons/hearing/XEH_postInit.sqf]: ?

File: addons/hearing/XEH_postInit.sqf [hunk 3]
```diff
@@ -35,7 +35,7 @@ GVAR(lastPlayerVehicle) = objNull;
             TRACE_2("removed veh eh",_firedEH,GVAR(lastPlayerVehicle));
         };
         if ((!isNull _vehicle) && {_player != _vehicle}) then {
-            private _firedEH = _vehicle addEventHandler ["FiredNear", LINKFUNC(firedNear)];
+            private _firedEH = _vehicle addEventHandler ["FiredNear", {call (firedNear)}];
```
Comment (commy2) [inline addons/hearing/XEH_postInit.sqf]: ?

**Comment by dedmen** [comment]
For reference: https://feedback.bistudio.com/T123355

**Pull request merged**